docs/typeChanges.md

```diff
@@ -1,5 +1,149 @@
+
+```json
+{
+    "minmax": [
+        104,
+        null
+    ],
+    "types": {
+        "newType": {
+          "someId": "u64",
+        }
+    }
+},
+```
+
+When the types.json file has been modified, it needs to be updated in the helm chart as well.
+
+```sh
+kubectl create configmap indexer-config --from-file=./typesBundle.json --dry-run=client --output=yaml > chart/templates/indexer-config.yaml
+```
+
+Also increment the indexer chart version `indexer/chart/chart.yaml`
+
+### 3: Run typegen exploration
+
+`cd typegenLocal`
+
+Copy over the typesBundle.json changes to `typegenLocal/typesBundle.json`.
+
+Once the typesBundle is modified, check if changes are required for generating types in `typegen.json`
+
+If you for example need to track a new event you need to edit the events declaration:
+
+```json
+  "events": [
+    "balances.Transfer",
+    "tfgridModule.EntityStored",
+    "tfgridModule.EntityUpdated",
+    "tfgridModule.EntityDeleted",
+    ...
+    "someModule.newEvent"
+    ...
+```
+
+Once you have done this, it's time to generate the types.
+
+```
+rm tfchainVersions.json
+```
+
+First, explore new types from the metadata:
+
+```
+npx squid-substrate-metadata-explorer \
+                --chain ws://localhost:9944 \
+                --out tfchainVersions.json
+```
+
+Then, run the type generation:
+
+```
+npx squid-substrate-typegen typegen.json
+```
+
+Notice it will re-generate following files:
+
+- src/typesLocal/events.js
+- src/typesLocal/$SPECVERSION.js
+
+### 4: Add new types / event to the main events definitions
+
+All events / types should be moved to `src/types` because we cannot generate types based on 1 chain because this code served more than 1 chain.
+
+At the inception of this project we generated `src/types` from devnet chain, but since we resetted devnet and we don't incrementally upgrade every network
+at all times, we stopped generating `src/types` and started modifiying them manually. While this is not really easy to develop against, it's the only solution for us.
+
+So we need to extend the generated src/types with incremental type changes we make over time.
+
+Steps:
+
+- move new $specVersion.ts file to `src/types`
+- copy changes from `src/typesLocal/events.ts` to `src/types/events.ts` (see which events were added / modified and only copy those over)
+
+### 5: Modify graphql schema
+
+Make changes to the graphql schema but prepare the database first:
+
+```
+yarn db:up
+yarn process
+```
+
+The process will likely crash or stop but the db is initialised with the current schema. Now we can edit the graphql schema file: `schema.graphql`.
+
+Once changes are done to the schema, you can create a database migration:
+
+```
+yarn codegen
+yarn build
+yarn db:create-migration
+```
+
+### 6: Create / extend mappers
+
+Make the corresponding changes in `src/processor.ts` and mappers `src/mappings/..` for the type changes that were made
```

Comment:
I don't see changes on those files in this PR but I guess this is because we don't modify existing types, we create new ones.

Reply:
Ah ok, I ll do it ;)